src/mbgl/renderer/render_source.hpp

```diff
@@ -93,6 +93,17 @@ class RenderSource : protected TileObserver {
         return {};
     }
 
+    virtual void setFeatureState(const optional<std::string>&,
+                                const std::string&,
+                                const style::conversion::Convertible&) {}
+
+    virtual PropertyMap getFeatureState(const optional<std::string>&,
+                                        const std::string&) const { return PropertyMap(); }
```

Comment:
`virtual FeatureState& ...`
inside impl, create static empty map and then return it?